app/assets/src/components/views/PipelineViz.jsx

```diff
@@ -25,29 +22,112 @@ class PipelineViz extends React.Component {
       ...(props.admin ? ["Experimental"] : []),
     ];
 
+    this.pipelineVersion = this.props.stageResults.pipeline_version;
+    this.stagesData = this.getStagesData();
+    this.graphs = [];
+    this.graphContainers = [];
+
     this.state = {
       stagesOpened: [true, true, true, true],
       zoom: 1,
+      sidebarVisible: false,
+      sidebarParams: {},
     };
   }
 
   componentDidMount() {
     this.drawGraphs();
   }
 
-  handleMouseWheelZoom = e => {
-    const { zoomChangeInterval } = this.props;
-    const zoomChange = (e.deltaY < 0 ? 1 : -1) * zoomChangeInterval;
-    this.setState({ zoom: this.state.zoom + zoomChange });
-  };
+  getStagesData() {
+    // TODO(ezhong): Include file download urls once passed up from backend.
+    const stageResults = this.stageResultsWithModifiedStepNames();
+    const filePathToOutputStep = this.generateFilePathToOutputStep();
+
+    const stages = Object.keys(stageResults).map(stageName => {
+      const rawStageData = stageResults[stageName];
+      const steps = rawStageData.steps.map(step => {
+        const name = step.class;
+
+        const inputInfo = step.in
+          .map(inTarget => {
```

Comment:
Did you indent this or was automatic? 
Maybe try to put the maps on the same line as the subject to reduce the levels of indentation as it makes it hard to read.

Reply:
It was automatic — not sure how to override it